docs/recipe_development.rst

```diff
@@ -0,0 +1,132 @@
+==============
+Recipe Development
+==============
+
+mBuild encourages users to subclass ``Compound`` to disseminate reproducible scripts used for building their systems.
+Sharing the script used to construct the initial configuration and apply force field parameters (if applicable) is a great step towards increasing the reproducibility of the system initialization step.
+
+However, there may be cases where those subclasses and/or scripts can be generalized to support a broad range of systems.
+Such subclasses would be a valuable resource for many researchers, and might justify development of a Python package that could be distributed to the community.
+
+
+mBuild has been developed with this in mind, in the form of a plug-in recipe system.
+Detailed below are the specifications of this system, how to convert your own Python project into an mBuild-discoverable plug-in, and an example.
+
+Entry Points
+--------
+
+The basis of the recipe system in mBuild relies on `setuptools.entry_points <https://packaging.python.org/guides/creating-and-discovering-plugins/#using-package-metadata>`_.
+This allows other packages to register themselves with the ``entry_point`` group we defined in mBuild, so they are accessible through the ``mbuild.recipes.my_recipe_foo`` location.
+In this case, ``my_recipe_foo`` would be the recipe you want to share.
+You can install many mBuild recipes, and only have to import mBuild to have access to them all!
+It is not necessary to use calls like this: ``from my_foo_project import my_recipe_foo``.
+The call ``import mbuild`` discovers all recipes that fit the ``entry_point`` group specification and makes them available under ``mbuild.recipes``. 
+
+Registering a Recipe
+________
+
+
+In the case that a user already has a Python project set up with a structure similar to the layout below:
```

Comment:
The next sentence is complete so this should be a complete sentence as well, I think?
```suggestion
Here we consider the case that a user already has a Python project set up with a structure similar to the layout below.
```